SendGrid/SendGridMail/Transport/Web.cs

```diff
@@ -87,7 +87,7 @@ public void Deliver(ISendGrid message)
 			AttachFormParams(message, content);
 			AttachFiles(message, content);
 			var response = await client.PostAsync(Endpoint + ".xml", content);
-			await CheckForErrorsAsync(response);
+            await ErrorChecker.CheckForErrorsAsync(response);

```

Comment:
You are absolutely right. I should have analysed more thoroughly.

The solution I came up is to execute the verification method synchronously after the content is available. Simplified snippet:

```csharp
    class Program
    {
        static void Main(string[] args)
        {
            DeliverAsync().Wait();
        }

        static async Task DeliverAsync()
        {
            using (var client = new HttpClient())
            {
                var response = await client.GetAsync("https://www.google.com").ConfigureAwait(false);
                var content = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
                checkResponse(response, content);
            }
        }

        private static void checkResponse(HttpResponseMessage response, string content)
        {
            if (content.Length != 5)
            {
                throw new Exception("Too short");
            }
            Console.WriteLine(content.Length);
        }
    }
```